From db3113aa935991ef25e40a2b0b7f22bb79f80a31 Mon Sep 17 00:00:00 2001
From: Patrick Rudolph <patrick.rudolph@9elements.com>
Date: Fri, 13 Oct 2023 14:18:46 +0200
Subject: [PATCH 1/3] meson: Fix meson options

Include config.h into all source files to make sure the #ifdef
directives used in code to evaluate defines from config.h.

Change-Id: I0f6a3aa750abdf7b7c263d05f0551ee481b98d51
Signed-off-by: Patrick Rudolph <patrick.rudolph@9elements.com>
---
 meson.build | 1 +
 1 file changed, 1 insertion(+)

diff --git a/meson.build b/meson.build
index 943cd37..9d22d76 100644
--- a/meson.build
+++ b/meson.build
@@ -21,6 +21,7 @@ conf_data.set('STRICT_FAILSAFE_PWM', get_option('strict-failsafe-pwm'))
 configure_file(output: 'config.h',
     configuration: conf_data
 )
+add_project_arguments('-include', 'config.h', language : 'cpp')
 
 if get_option('oe-sdk').enabled()
     OECORE_TARGET_SYSROOT = run_command('sh', '-c', 'echo $OECORE_TARGET_SYSROOT').stdout().strip()
-- 
2.41.0

